src/live_restore/live_restore_fs.c

```diff
@@ -1178,68 +1182,104 @@ __live_restore_fs_open_in_destination(WT_LIVE_RESTORE_FS *lr_fs, WT_SESSION_IMPL
 }
 
 /*
- * __live_restore_fs_open_file --
- *     Open a live restore file handle. This will: - If the file exists in the source, open it in
- *     both. - If it doesn't exist it'll only open it in the destination.
+ * __live_restore_setup_lr_fh_directory --
+ *     Populate a live restore file handle for a directory. Directories have special handling. If
+ *     they don't exist in the destination they'll be created immediately (but not their contents)
+ *     and immediately marked as complete. WiredTiger will never create or destroy a directory so we
+ *     don't need to think about tombstones.
  */
 static int
-__live_restore_fs_open_file(WT_FILE_SYSTEM *fs, WT_SESSION *wt_session, const char *name,
-  WT_FS_OPEN_FILE_TYPE file_type, uint32_t flags, WT_FILE_HANDLE **file_handlep)
+__live_restore_setup_lr_fh_directory(WT_SESSION_IMPL *session, WT_LIVE_RESTORE_FS *lr_fs,
+  const char *name, uint32_t flags, WT_LIVE_RESTORE_FILE_HANDLE *lr_fhp)
 {
     WT_DECL_RET;
-    WT_LIVE_RESTORE_FILE_HANDLE *lr_fh;
-    WT_LIVE_RESTORE_FS *lr_fs;
-    WT_LIVE_RESTORE_FS_LAYER_TYPE which;
-    WT_SESSION_IMPL *session;
-    bool dest_exist, source_exist, have_tombstone, readonly;
+    char *dest_folder_path = NULL;
+    bool dest_exist = false, source_exist = false;
 
-    session = (WT_SESSION_IMPL *)wt_session;
-    lr_fs = (WT_LIVE_RESTORE_FS *)fs;
+    WT_RET_NOTFOUND_OK(
+      __live_restore_fs_has_file(lr_fs, &lr_fs->destination, session, name, &dest_exist));
+    WT_RET_NOTFOUND_OK(
+      __live_restore_fs_has_file(lr_fs, &lr_fs->source, session, name, &source_exist));
 
-    dest_exist = source_exist = false;
-    lr_fh = NULL;
-    have_tombstone = false;
-    WT_UNUSED(have_tombstone);
-    readonly = LF_ISSET(WT_FS_OPEN_READONLY);
-    WT_UNUSED(readonly);
-    WT_UNUSED(which);
+    if (!dest_exist && !source_exist && !LF_ISSET(WT_FS_OPEN_CREATE))
+        WT_RET_MSG(session, ENOENT, "Directory %s does not exist in source or destination", name);
 
-    /* FIXME-WT-13808 Handle WT_FS_OPEN_FILE_TYPE_DIRECTORY */
+    if (!dest_exist) {
+        /* The directory doesn't exist in the destination yet. We need to create it in all cases. */
 
-    /* Set up the file handle. */
-    WT_ERR(__wt_calloc_one(session, &lr_fh));
-    WT_ERR(__wt_strdup(session, name, &lr_fh->iface.name));
-    lr_fh->iface.file_system = fs;
-    lr_fh->file_type = file_type;
+        WT_RET(__live_restore_create_file_path(
+          session, &lr_fs->destination, (char *)name, &dest_folder_path));
 
-    WT_ERR(__wt_rwlock_init(session, &lr_fh->ext_lock));
+        /*
+         * We can't create directories with a WT_FS_OPEN_CREATE call. Instead we do it manually.
```

Comment:
Why not?

Reply:
There's no support for this in `__posix_open_file`. I guess we could add it but I'd like to avoid modifying the posix layer where we can.

Reply:
Can you add that detail to the comment?